schemas/iso19139/src/main/plugin/iso19139/extract-relations.xsl

```diff
@@ -75,6 +77,12 @@
                   else if ($mainLanguage) then $mainLanguage else $lang}">
         <xsl:value-of select="."/>
       </value>
+
+      <xsl:if test="gmx:Anchor">
```

Comment:
I don't think this can work. If you have no `gco:CharacterString|gmd:PT_FreeText/*/gmd:LocalisedCharacterString` you don't enter the for each. And even if you do, I think the context is the element so to reach the Anchor you need ../gmx:Anchor.

Anyway, I would simplify this by:
```

    <xsl:for-each select="gco:CharacterString|gmx:Anchor|
                          gmd:PT_FreeText/*/gmd:LocalisedCharacterString">
      <xsl:variable name="localeId"
                    select="substring-after(@locale, '#')"/>

      <value lang="{if (@locale)
                  then ancestor::metadata/*[@gco:isoType='gmd:MD_Metadata' or name()='gmd:MD_Metadata']/gmd:locale/*[@id = $localeId]/gmd:languageCode/*/@codeListValue
                  else if ($mainLanguage) then $mainLanguage else $lang}">
        <xsl:copy-of select="@xlink:href"/>
        <xsl:value-of select="."/>
      </value>
    </xsl:for-each>
```
which would be easier. not sure how to test on the client side ?

Reply:
You change looks fine, and indeed the `xsl:if` should be outside of the loop. I merged from other project where was fine and mess up with the merge for the PR